lib/serialization/serializationOptions.ts

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+import { TypeSpec } from "./typeSpec";
+import { CompositeType } from "./compositeSpec";
+
+/**
+ * Options that can be passed to a serialize() function.
+ */
+export interface SerializationOptions {
+  /**
+   * The type of output that will be produced.
+   */
+  outputType?: SerializationOutputType;
+
+  /**
+   * Whether or not serialization will follow strict type-checking. If strict type-checking is used,
+   * then an Error will be thrown if a value doesn't match the provided TypeSpec's expected types.
+   */
+  serializationStrictTypeChecking?: boolean;
```

Comment:
Is the idea here that Azure devs might opt in to these strictness options based on their personal preference?

Reply:
Personal preference, or for debugging. I actually just used these options this morning to detect a problem with an operation spec.
@amarzavery mentioned that sometimes resource provider teams will return types that are different than what the swagger indicates they'll be. This shouldn't happen, but sometimes it does. In those cases, it's better to allow the customer to specify that they want the deserialization to be looser so that they can get the response and work with it, even if it doesn't match the specification perfectly.